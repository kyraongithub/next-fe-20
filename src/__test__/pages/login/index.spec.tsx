import LoginPage from '@/pages/login';
import { fireEvent, render } from '@testing-library/react';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe('Login Page', () => {
  beforeEach(async () => {
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');

    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    }));
  });
  it('should render login page', () => {
    const { container } = render(<LoginPage />);
    expect(container).toMatchSnapshot();
  });

  it('should click login button', () => {
    const { getByText, container } = render(<LoginPage />);
    const button = getByText('Login');
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
  it('should change input email value', () => {
    const { getByPlaceholderText } = render(<LoginPage />);
    const input = getByPlaceholderText('email');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input).toHaveValue('test');
  });
  it('should change input password value', () => {
    const { getByPlaceholderText } = render(<LoginPage />);
    const input = getByPlaceholderText('password');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input).toHaveValue('test');
  });
});
