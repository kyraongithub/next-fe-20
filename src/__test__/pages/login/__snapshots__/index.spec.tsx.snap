// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should click login button 1`] = `
<div>
  <div>
    <input
      placeholder="email"
      type="text"
    />
    <input
      placeholder="password"
      type="password"
    />
    <button>
      Login
    </button>
  </div>
</div>
`;

exports[`Login Page should render login page 1`] = `
<div>
  <div>
    <input
      placeholder="email"
      type="text"
    />
    <input
      placeholder="password"
      type="password"
    />
    <button>
      Login
    </button>
  </div>
</div>
`;
